files/check_zpool_scrub

```diff
@@ -0,0 +1,527 @@
+
+##
+# Extract the time to go from the zpool status output.
+#
+# Parameters:
+# $1: POOL: Name of the pool
+#
+# Responsible for the output is this line:
+#   (void) printf(gettext(", %lluh%um to go\n"),
+# in the ZFS source code.
+# https://github.com/openzfs/openzfs/blob/ed81aacb0d0fcbf7e0c0745ea4556655050c26bf/usr/src/cmd/zpool/zpool_main.c#L4441
+#
+# For example:
+# Input:
+#   ...
+#   scan: scrub in progress since Sun Aug 13 00:24:02 2017
+#     7,34T scanned out of 10,1T at 57,4M/s, 14h12m to go
+#     0 repaired, 72,38% done
+#   ...
+# Output:
+#     14h12m
+#
+# Other solutions:
+#
+# Not working on FreeBSD:
+#   grep -P -o '(?<=, )[[:digit:]]*h[[:digit:]]*m(?= to go)'
+#
+# Naive solution:
+#   grep ' to go' |
+#   awk '{ print $8 }'
+##
+time_grep() {
+	zpool status "$1" | grep ' to go' | sed -E 's/^.*, (.*h.*m) to go.*$/\1/g' || true
+}
+
+time_to_min() {
+	local HOURS MIN
+	HOURS=$(echo "$1" | sed 's/h.*//')
+	MIN=$(echo "$1" | sed 's/.*h//' | sed 's/m//')
+	echo $((HOURS * 60 + MIN))
+}
+
+_time() {
+	local GREP
+	GREP="$(time_grep "$1")"
+	if [ -n "$GREP" ]; then
+		time_to_min "$GREP"
+	else
+		echo 0
+	fi
+}
+
+########################################################################
+# Main functions
+########################################################################
+
+##
+# Check one ZFS pool.
+#
+# Parameters:
+#   $1: The name of the pool
+##
+check_one_pool() {
+	local POOL="$1" NOW LAST_SCRUB DIFF PROGRESS SPEED TIME
+
+	if ! zpool status "$POOL" > /dev/null 2>&1; then
+		checkOutput="'${POOL}' is no ZFS pool."
+		checkState=UNKNOWN
+		return
+	fi
+
+	NOW=$(now_to_timestamp)
+	LAST_SCRUB=$(last_scrub_to_timestamp "$POOL")
+
+	if [ -z "$LAST_SCRUB" ]; then
+		checkOutput="The pool has never had a scrub."
+		checkState=UNKNOWN
+		return
+	fi
+
+	DIFF=$((NOW - LAST_SCRUB))
+	PROGRESS=$(progress "$POOL")
+	SPEED="$(speed "$POOL")"
+	TIME="$(_time "$POOL")"
+
+	checkState=UNKNOWN
+
+	if [ "$DIFF" -gt "$OPT_CRITICAL" ]; then
+		checkState=CRITICAL
+	elif [ "$DIFF" -gt "$OPT_WARNING" ]; then
+		checkState=WARNING
+	else
+		checkState=OK
+	fi
+
+	checkPerfdata="$checkPerfdata \
+$(performance_data_one_pool \
+		"${POOL}" "${DIFF}" "${PROGRESS}" "${SPEED}" "${TIME}")"
+	checkOutput="${checkOutput}The last scrub on zpool '${POOL}' was \
+performed on $(timestamp_to_datetime $LAST_SCRUB)."
+
+}
+
+check_multiple_pools() {
+	local POOL GLOBAL_STATE=OK
+
+	for POOL in $@; do
+		check_one_pool "$POOL"
+		# If one pool is critical the global state turns to
+		# critical.
+		if [ "${checkState}" = "CRITICAL" ]; then
+			GLOBAL_STATE=CRITICAL
+		# warning
+		elif [ "${checkState}" = "WARNING" ] && \
+		     ! [ ${GLOBAL_STATE} = "CRITICAL" ]; then
+			GLOBAL_STATE=WARNING
+		# unknown
+		elif [ "${checkState}" = "UNKNOWN" ] && \
+		     ! [ ${GLOBAL_STATE} = "WARNING" ] && \
+		     ! [ ${GLOBAL_STATE} = "CRITICAL" ]; then
+			GLOBAL_STATE=UNKNOWN
+		fi
+	done
+	checkState="$GLOBAL_STATE"
+}
+
+_getopts $@
```

Comment:
That's a strange name

Reply:
Guess what, I'm sure you know that `getopts` is a bash directive, which is why I can't use it as a function name.